File: app/GOExercise/GOExercise/Models/WorkoutCategory.swift
import Foundation

enum WorkoutCategory: String, Codable, CaseIterable, Identifiable, Sendable {
    // 並び順 = ピッカー/チップの表示順。筋トレを先頭 (記録のデフォルト) に。
    // rawValue は変えないので保存データには影響しない。
    case strength
    case cardio
    case yoga
    case stretch
    case fasciaRelease
    case other

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .cardio: "有酸素"
        case .strength: "筋トレ"
        case .yoga: "ヨガ"
        case .stretch: "ストレッチ"
        case .fasciaRelease: "筋膜リリース"
        case .other: "その他"
        }
    }

    var symbolName: String {
        switch self {
        case .cardio: "figure.run"
        case .strength: "dumbbell.fill"
        case .yoga: "figure.mind.and.body"
        case .stretch: "figure.cooldown"
        case .fasciaRelease: "figure.flexibility"
        case .other: "sparkles"
        }
    }
}

File: app-android/app/src/main/java/com/goexercise/app/domain/WorkoutCategory.kt
package com.goexercise.app.domain

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * 運動カテゴリ。iOS `WorkoutCategory.swift` の 1:1 移植。
 * 並び順 = ピッカー/チップ表示順(筋トレ先頭=記録デフォルト)。`rawValue` は iOS と一致させ
 * 保存データ(Supabase/Room)で相互運用できるようにする(変更しない)。
 * JSON シリアライズも rawValue を使う(@SerialName)ことで iOS と同じ表現にする。
 */
@Serializable
enum class WorkoutCategory(val rawValue: String, val displayName: String) {
    @SerialName("strength") Strength("strength", "筋トレ"),
    @SerialName("cardio") Cardio("cardio", "有酸素"),
    @SerialName("yoga") Yoga("yoga", "ヨガ"),
    @SerialName("stretch") Stretch("stretch", "ストレッチ"),
    @SerialName("fasciaRelease") FasciaRelease("fasciaRelease", "筋膜リリース"),
    @SerialName("other") Other("other", "その他");

    companion object {
        /** 不明な rawValue は other にフォールバック(iOS の `WorkoutCategory(rawValue:) ?? .other`)。 */
        fun fromRaw(raw: String?): WorkoutCategory =
            entries.firstOrNull { it.rawValue == raw } ?: Other
    }
}
